### DISCUSSION TITLE
How to produce enterprise reports using quarto and jupyter

### DISCUSSION THREAD

**Discussion by a0th** [quarto-dev/quarto-cli#7695] [Q&A]
Hey guys, as a data scientist I need to generate a lot of reports and I spend a quite some time in Jupyter. 
What Id like to do is to generate enterprise-y looking reports in PDF for the stakeholders from a jupyter notebook.
Ive used latex in college, but thats definitely not my forte, and I dont really like the defaults, so I am looking for some directions.

I tried using the html rendering with weasyprint and pagedjs, but they still fall short in some details, specifically:
1. no page numbers
2. chopping charts and tables when they clearly should be in the next page
3. the `<pagebreak>` shortcode doesnt work properly

So html to pdf feels unreliable because every tiny change requires a full review of the whole document.

While using latex, most of these things work, but on the other hand, the table format feels like a huge downgrade. I can make the tables look decent with the .style method in pandas, and default quarto themes also work very well while using HTML. But in PDF, I strongly dislike the feel of the tables. The style method is also used to format numbers, like turning 0.001001 into 0.1%, but that is ignored by latex, because style produces html.

From this point, I have three options:

1. Stick to HTML to PDF, keep using CSS which is very easy to find help for, and try to find a solution for the shortcomings
2. Try to learn latex, although the philosophy design feels very outdated compared to html authoring.
3. Find another solution to convert jupyter into PDF.

For the reference, [here's what I consider an enterprise report](https://www.hedgeinvest.com.br/arquivos/HFOF/Outros_Materiais/2021_09_21_HFOF_Resultados_e_Perspectivas.pdf).

What are you guys thoughts on this? Any help is appreciated!

**Comment by mcanouil** [comment]
Have you considered using Typst (part of Quarto 1.4)?

And because the example you should is basically slides, have you try to use `revealjs` format which has an "export to pdf" option which actually don't mess the HTML (unless you have very advanced HTML stuff in it).
I would create a `revealjs` template extension and use it to make "reports" in HTML then use the export feature or use tools such as Phantom.js (or derivated tools) to make a PDF.

**Comment by a0th** [comment]
Thanks a lot for the suggestions! I'm checking them out.
Is there a way to reduce the font size in the slides using a quarto configuration other than the `smaller`?

**Reply by mcanouil** [reply]
Note that most options are listed and searchable in the documentation: https://quarto.org/docs/reference/formats/presentations/revealjs.html

Theming, layout, etc. are I believe extensively detailed in the guides: https://quarto.org/docs/presentations/revealjs/

In case, you did not know, Reveal.js is a JavaScript library to have HTML based slides thus CSS/SCSS works in it.
From the guide, you can find two SASS variables that control font size and the font size associated with "smaller", see <https://quarto.org/docs/presentations/revealjs/themes.html#fonts>.
